worldwind-gt-geopkg/src/main/java/gov/nasa/worldwind/geopkg/mosaic/GeoPackageReader.java

```diff
@@ -90,21 +90,44 @@
     protected File sourceFile;
 
     protected Map<String, TileEntry> tiles = new HashMap<String, TileEntry>();
+    
+    // TODO: store these gridRanges in the TileEntry class
+    protected Map<String, GridEnvelope2D> gridRanges = new HashMap<String, GridEnvelope2D>();
 
     public GeoPackageReader(Object source, Hints hints) throws IOException {
         coverageFactory = CoverageFactoryFinder.getGridCoverageFactory(this.hints);
         sourceFile = GeoPackageFormat.getFileFromSource(source);
         GeoPackage file = new GeoPackage(sourceFile);
 
         try {
-            // Set the coverage name to the name of the the first raster tileset.
             coverageName = null;   // default value was "geotools_coverage"
-            for (TileEntry tile : file.tiles()) {
-                tiles.put(tile.getTableName(), tile);
+            for (TileEntry tileset : file.tiles()) {
+                // Map the tileset to the coverage name (table name)
+                tiles.put(tileset.getTableName(), tileset);
+
+                // Set the default coverage name to the name of the the first raster tileset.
                 if (coverageName == null) {
-                    coverageName = tile.getTableName();
+                    coverageName = tileset.getTableName();
                 }
+
+                // TODO: compute the gridRange in the GeoPackage class
+                List<TileMatrix> matricies = tileset.getTileMatricies();
+                TileMatrix matrix = matricies.get(matricies.size() - 1);
+                int maxZoomLevel = matrix.getZoomLevel();
```

Comment:
There are two aspects of the GeoPackage spec that could be problematic here:

1. GeoPackage does not stipulate that tile matrices are ordered, which could invalidate the assumption that the last TileMatrix represents the maximum zoom level. Does the method TileEntry.getTileMatrices provide this guarantee?

2. GeoPackage allows the definition of tile matrices that have no corresponding tile data in the file. For example, a GeoPackage could define 17 logical levels of tile matrices, yet provide tile data for only levels 5-10. We have test files that do this. In WorldWind Android, I queried the GeoPackage to identify the TileMatrix with the largest zoom level that also had at least one corresponding tile in the tile table.